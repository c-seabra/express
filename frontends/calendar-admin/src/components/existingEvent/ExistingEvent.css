/* wrapper */
.wrapper {
  padding: 0 1.5rem 1.5rem;
}

/* END wrapper */

/* existingEventItem */
.item {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  margin-bottom: 1.5rem
}

.itemIcon {
  width: 20px;
}

.itemIcon svg {
  width: 100%;
}

.itemChildren {
  padding: 0;
  margin: 0;
  width: calc(100% - 42px);
}

/* END existingEventItem */

/* Title & date */
.title {
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0 0 .25rem;
}

.date {
  margin: 0;
  font-size: .875rem;
}

/* END Title & date */

/* userRsvp styles */
.userRsvp {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #3E3D44;
  padding: 1.5rem;
  color: white;
  height: 85px;
}

.userRsvp_button {
  color: #fff;
}

/* END userRsvp styles */

/* Delete overlay */
.overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  background: rgba(255, 255, 255, .8);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 2rem 1.5rem;
}

.overlayButtons {
  display: flex;
  align-items: center;
  justify-content: center;
}

/* Overlay buttons */
.overlayButton {
  color: #3E3D44;
  border: 1px solid #cecece;
}

.overlayButton:hover {
  background: #3E3D44;
}

.overlayButton:first-child {
  margin-left: 0;
}

.overlayButton_yes {
  background: #FA7E23;
  border-color: #FA7E23;
  color: white;
}

.overlayButton_yes:hover {
  background: red;
}

/* END Overlay buttons */
/* END Delete overlay */


.formWrap {
  position: fixed;
  top: 0;
  height: 100%;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  background: white;
  padding-top: 0;
}
