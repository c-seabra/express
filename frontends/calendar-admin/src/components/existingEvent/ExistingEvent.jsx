import ButtonComponent from '@websummit/components/src/atoms/Button';
import moment from 'moment';
import React, { useContext, useEffect, useState } from 'react';

import AvatarList from '../avatarList/AvatarList';
import Button from '../button/Button';
import { DetailsContext } from '../calendar/Context';
import EditEvent from '../editEvent/EditEvent';
import { TopButtons } from '../popup/Popup.styled';
import BinIcon from '../svgs/Bin';
import CloseIcon from '../svgs/Close';
import DescriptionIcon from '../svgs/Description';
import GroupIcon from '../svgs/Group';
import MeetingIcon from '../svgs/Meeting';
import OvalSquare from '../svgs/OvalSquare';
import PenIcon from '../svgs/Pen';
import {
  Date,
  FormWrap,
  Overlay,
  OverlayButton,
  OverlayButtons,
  PopupContainer,
  PopupHeading,
  Title,
  UserRsvp,
  Wrapper,
} from './ExistingEvent.styled';
import ExistingEventItem from './ExistingEventItem';

const ExistingEvent = ({ event, close_popup, formats }) => {
  const {
    id,
    title,
    description,
    location,
    invitations,
    starts_at,
    ends_at,
    event_format_id,
    can_modify,
    organizer,
  } = event;

  const {
    currentUserId,
    locations,
    rsvps,
    responseStatuses,
    onUpdateEventInvitationResponse,
    onDeleteEvent,
  } = useContext(DetailsContext);

  const [deletePopupActive, setDeletePopupActive] = useState(false);
  const [editPopupActive, setEditPopupActive] = useState(false);
  const [locationName, setLocationName] = useState();
  const [currentUserInvitation, setCurrentUserInvitation] = useState();
  const [eventTime, setEventTime] = useState();

  const formatDate = (startsAt, endsAt) => {
    const dateFormatted = `${moment(startsAt).format(
      'dddd, DD MMMM - HH:mm',
    )}-${moment(endsAt).format('HH:mm')}`;

    setEventTime(dateFormatted);
  };

  useEffect(() => {
    // get location name
    location && setLocationName(getLocationName(location));

    // get current user response
    // so we can set RSVP buttons
    const invitation =
      invitations &&
      invitations.find((invite) => invite.invitee.id === currentUserId);
    setCurrentUserInvitation(invitation);

    // Format the date to desired layout
    formatDate(starts_at.toString(), ends_at.toString());
  }, [location && location.id, location && location.name, invitations, rsvps]);

  const handleDelete = () => {
    // open popup overlay
    setDeletePopupActive(true);
  };
  const handleDeleteResponse = (deleteResponse) => {
    if (deleteResponse) onDeleteEvent(id);
    setDeletePopupActive(false);
  };

  const handleEdit = () => {
    setEditPopupActive(!editPopupActive);
  };

  const getLocationName = () => {
    let currentLocationName;

    if (location.type === 'text') {
      currentLocationName = location.name;
    } else {
      const currentLocation =
        locations &&
        locations.find(
          (confLocation) =>
            confLocation.id === location.id || confLocation.id === location,
        );
      currentLocationName = currentLocation && currentLocation.name;
    }

    return currentLocationName;
  };

  return (
    <div className="existingEvent">
      <TopButtons>
        {can_modify && (
          <>
            {!editPopupActive && (
              <Button className="topButton" onBtnClick={handleEdit}>
                <PenIcon />
              </Button>
            )}
            <Button className="topButton" onBtnClick={handleDelete}>
              <BinIcon />
            </Button>
          </>
        )}
        <Button className="topButton" onBtnClick={close_popup}>
          <CloseIcon />
        </Button>
      </TopButtons>

      <Wrapper>
        <ExistingEventItem icon={<OvalSquare iconColor="#f40" />}>
          <Title>{title}</Title>
          <Date>{eventTime}</Date>
        </ExistingEventItem>

        {locationName && (
          <ExistingEventItem icon={<MeetingIcon />}>
            {locationName}
          </ExistingEventItem>
        )}

        <ExistingEventItem icon={<GroupIcon />}>
          {rsvps && rsvps.length ? (
            <AvatarList
              avatarList={rsvps}
              organizerId={organizer ? organizer.id : undefined}
              listType="rsvps"
            />
          ) : (
            'Loading...'
          )}
        </ExistingEventItem>

        {description && (
          <ExistingEventItem icon={<DescriptionIcon />}>
            {description}
          </ExistingEventItem>
        )}
      </Wrapper>

      {editPopupActive && (
        <FormWrap>
          <PopupContainer>
            <PopupHeading>Edit Event</PopupHeading>
            <EditEvent
              endsAt={ends_at}
              eventFormatId={event_format_id}
              eventId={id}
              organizerId={organizer ? organizer.id : undefined}
              setEditPopupActive={setEditPopupActive}
              startsAt={starts_at}
              {...{
                description,
                location,
                rsvps,
                title,
              }}
              formats={formats}
            />
          </PopupContainer>
        </FormWrap>
      )}

      {deletePopupActive && (
        <Overlay>
          <h3>Are you sure you want to delete this event?</h3>
          <OverlayButtons>
            <ButtonComponent
              type="primary"
              onClick={() => handleDeleteResponse(true)}
            >
              Yes
            </ButtonComponent>

            <ButtonComponent
              type="error"
              onClick={() => handleDeleteResponse(false)}
            >
              No
            </ButtonComponent>
          </OverlayButtons>
        </Overlay>
      )}

      <UserRsvp>
        <div>Going?</div>
        <div>
          {currentUserInvitation &&
            currentUserInvitation.valid_response_status_ids.map((response) => {
              const currentResponse =
                responseStatuses &&
                responseStatuses.find(
                  (responseStatus) => responseStatus.id === response,
                );
              return (
                <Button
                  key={response}
                  className={`userRsvp_button ${
                    currentUserInvitation.response.response_status === response
                      ? 'button_active'
                      : ''
                  }`}
                  onBtnClick={() =>
                    currentUserInvitation.response.response_status !== response
                      ? onUpdateEventInvitationResponse(id, response)
                      : ''
                  }
                >
                  {currentResponse && currentResponse.label}
                </Button>
              );
            })}
        </div>
      </UserRsvp>
    </div>
  );
};
export default ExistingEvent;
