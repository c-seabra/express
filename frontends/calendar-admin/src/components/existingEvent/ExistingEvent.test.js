import React from 'react';
import renderer from 'react-test-renderer';

import ExistingEvent from './ExistingEvent';

describe('ExistingEvent', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    React.useContext = jest.fn()
    useContextMock = React.useContext;
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  test('renders without props or context', () => {
    useContextMock.mockReturnValue({});
    const component = renderer.create(
      <ExistingEvent />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders with props, and no context', () => {
    useContextMock.mockReturnValue({});
    const component = renderer.create(
      <ExistingEvent
        title={'some title'}
        starts_at={'2019-04-11T06:15:06.078Z'}
        description={'catching up'}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders with props and context', () => {
    useContextMock.mockReturnValue({
      location: {
        name: 'mock location name',
        id: 1234
      },
      rsvps: [
        {
          invitation: { id: 123, response: { response_status: 'accepted' } },
          attendance: { data: { person: { first_name: 'Tomasz' } } }
        }
      ]
    });
    const component = renderer.create(
      <ExistingEvent
        title={'some title'}
        starts_at={'2019-04-11T06:15:06.078Z'}
        description={'catching up'}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
