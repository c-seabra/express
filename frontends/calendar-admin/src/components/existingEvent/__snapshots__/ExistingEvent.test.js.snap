// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExistingEvent renders with props and context 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <button
      className={undefined}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 15 15"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            Rectangle
          </title>
          <desc>
            Created with Sketch.
          </desc>
          <g
            fill="none"
            fillRule="evenodd"
            id="Symbols"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="#FA7E23"
              fillRule="nonzero"
              id="1:1-Private-Meeting-small"
              transform="translate(-23.000000, -64.000000)"
            >
              <rect
                fill="#f40"
                height="15"
                id="Rectangle"
                rx="4"
                width="15"
                x="23"
                y="64"
              />
            </g>
          </g>
        </svg>
      </div>
      <div
        className={undefined}
      >
        <h2
          className={undefined}
        >
          some title
        </h2>
        <p
          className={undefined}
        >
          Thursday, 4 April - 6:15-NaN:NaN
        </p>
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          aria-hidden="true"
          data-icon="users"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className={undefined}
      >
        <ul
          className={undefined}
        >
          <li>
            <div
              className={undefined}
            >
              <div
                className={undefined}
              >
                <object
                  data={undefined}
                  type="image/png"
                >
                  <img
                    alt="Invitee avatar"
                    src="https://s3-eu-west-1.amazonaws.com/assets.cilabs.com/images/person.svg"
                    style={
                      Object {
                        "borderColor": "transparent",
                      }
                    }
                  />
                </object>
                <svg
                  viewBox="0 0 10 9"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <clipPath
                      id="vrmva"
                    >
                      <path
                        d="M4.125 5.725L2.9 4.5l-.408.408 1.633 1.634 3.5-3.5-.408-.409z"
                      />
                    </clipPath>
                  </defs>
                  <g>
                    <g>
                      <g>
                        <path
                          d="M5 9a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                          fill="#f8992c"
                        />
                      </g>
                      <g>
                        <g />
                        <g
                          clipPath="url(#vrmva)"
                        >
                          <g>
                            <path
                              d="M1.5 1h7v7h-7z"
                              fill="#fff"
                            />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </svg>
              </div>
              <div
                className={undefined}
              >
                <p
                  className={undefined}
                >
                  Tomasz undefined
                </p>
                <p
                  className={undefined}
                >
                  Organizer
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          aria-hidden="true"
          data-icon="align-left"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.83 344h262.34A12.82 12.82 0 0 0 288 331.17v-22.34A12.82 12.82 0 0 0 275.17 296H12.83A12.82 12.82 0 0 0 0 308.83v22.34A12.82 12.82 0 0 0 12.83 344zm0-256h262.34A12.82 12.82 0 0 0 288 75.17V52.83A12.82 12.82 0 0 0 275.17 40H12.83A12.82 12.82 0 0 0 0 52.83v22.34A12.82 12.82 0 0 0 12.83 88zM432 168H16a16 16 0 0 0-16 16v16a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-16a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v16a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-16a16 16 0 0 0-16-16z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className={undefined}
      >
        catching up
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div>
      Going?
    </div>
    <div>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Yes
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        No
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Maybe
      </button>
    </div>
  </div>
</div>
`;

exports[`ExistingEvent renders with props, and no context 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <button
      className={undefined}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 15 15"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            Rectangle
          </title>
          <desc>
            Created with Sketch.
          </desc>
          <g
            fill="none"
            fillRule="evenodd"
            id="Symbols"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="#FA7E23"
              fillRule="nonzero"
              id="1:1-Private-Meeting-small"
              transform="translate(-23.000000, -64.000000)"
            >
              <rect
                fill="#f40"
                height="15"
                id="Rectangle"
                rx="4"
                width="15"
                x="23"
                y="64"
              />
            </g>
          </g>
        </svg>
      </div>
      <div
        className={undefined}
      >
        <h2
          className={undefined}
        >
          some title
        </h2>
        <p
          className={undefined}
        />
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          aria-hidden="true"
          data-icon="users"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className={undefined}
      >
        Loading...
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          aria-hidden="true"
          data-icon="align-left"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.83 344h262.34A12.82 12.82 0 0 0 288 331.17v-22.34A12.82 12.82 0 0 0 275.17 296H12.83A12.82 12.82 0 0 0 0 308.83v22.34A12.82 12.82 0 0 0 12.83 344zm0-256h262.34A12.82 12.82 0 0 0 288 75.17V52.83A12.82 12.82 0 0 0 275.17 40H12.83A12.82 12.82 0 0 0 0 52.83v22.34A12.82 12.82 0 0 0 12.83 88zM432 168H16a16 16 0 0 0-16 16v16a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-16a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v16a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-16a16 16 0 0 0-16-16z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className={undefined}
      >
        catching up
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div>
      Going?
    </div>
    <div>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Yes
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        No
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Maybe
      </button>
    </div>
  </div>
</div>
`;

exports[`ExistingEvent renders without props or context 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <button
      className={undefined}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 15 15"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <title>
            Rectangle
          </title>
          <desc>
            Created with Sketch.
          </desc>
          <g
            fill="none"
            fillRule="evenodd"
            id="Symbols"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="#FA7E23"
              fillRule="nonzero"
              id="1:1-Private-Meeting-small"
              transform="translate(-23.000000, -64.000000)"
            >
              <rect
                fill="#f40"
                height="15"
                id="Rectangle"
                rx="4"
                width="15"
                x="23"
                y="64"
              />
            </g>
          </g>
        </svg>
      </div>
      <div
        className={undefined}
      >
        <h2
          className={undefined}
        />
        <p
          className={undefined}
        />
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <svg
          aria-hidden="true"
          data-icon="users"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        className={undefined}
      >
        Loading...
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div>
      Going?
    </div>
    <div>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Yes
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        No
      </button>
      <button
        className="undefined undefined "
        onClick={[Function]}
      >
        Maybe
      </button>
    </div>
  </div>
</div>
`;
